### PR TITLE
Make ProductId comparable

### PR DESCRIPTION
**Pull request by jenniferboedker**
In order to properly sort the product identifier it must implement the comparable interface

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy [hunk 1]
```diff
@@ -124,4 +124,11 @@ class ProductId {
     String toString() {
         return "${type}_${uniqueId}"
     }
+
+    @Override
+    int compareTo(ProductId productId2) {
+        //if the types are equal compare by uniqueId
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy:130]: I would suggest using the equals method here to make sure it behaves as expected.
```suggestion
        //if the types are equal compare by uniqueId
        if(this.equals(productId2)) return 0
```

**Review by KochTobi — CHANGES_REQUESTED** [review]
I would suggest using the equals method as well in the compareTo method or adding the note "Note: this class has a natural ordering that is inconsistent with equals." to the JavaDoc

File: src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy (hunk 1 above)
Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy:130]: done :)

Comment (KochTobi) [reply src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy:130]: Sorry but that is not what i meant :) 
The operator in the type comparison is fine. What I meant is, that the productId compareTo method needs to be in sync with the equals method of ProductId

**Comment by jenniferboedker** [comment]
[Service Products: Default product sorting #574](https://trello.com/c/gaIZzYZN/716-service-products-default-product-sorting-574)

**Review by KochTobi — APPROVED** [review]
Looks good :+1:

**Pull request merged**